unittest/casadi/joints.cpp

```diff
@@ -390,6 +427,145 @@ struct TestADOnJoints
   { /* do nothing */
   }
 
+  template<typename Scalar, int Options>
+  static void test(const pinocchio::JointModelSplineTpl<Scalar, Options> & jmodel)
+  {
+    std::cout << "--" << std::endl;
+    std::cout << "jmodel: " << jmodel.shortname() << std::endl;
+
+    typedef casadi::SX AD_double;
+
+    typedef pinocchio::SE3Tpl<AD_double> SE3AD;
+    typedef pinocchio::MotionTpl<AD_double> MotionAD;
+    typedef pinocchio::SE3Tpl<double> SE3;
+    typedef pinocchio::MotionTpl<double> Motion;
+    typedef pinocchio::JointMotionSubspaceTpl<Eigen::Dynamic, double> JointMotionSubspaceXd;
+
+    typedef Eigen::Matrix<AD_double, Eigen::Dynamic, 1> VectorXAD;
+    typedef Eigen::Matrix<AD_double, 6, 1> Vector6AD;
+
+    typedef pinocchio::JointModelSplineTpl<Scalar, Options> JointModel;
+    typedef typename pinocchio::CastType<AD_double, JointModel>::type JointModelAD;
+    typedef typename JointModelAD::JointDataDerived JointDataAD;
+
+    typedef typename JointModelAD::ConfigVector_t ConfigVectorAD;
+
+    typedef typename JointModel::JointDataDerived JointData;
+    typedef typename JointModel::ConfigVector_t ConfigVector;
+    typedef typename JointModel::TangentVector_t TangentVector;
+
+    JointData jdata(jmodel.createData());
+    pinocchio::JointDataBase<JointData> & jdata_base = jdata;
+
+    JointModelAD jmodel_ad = jmodel.template cast<AD_double>();
+    JointDataAD jdata_ad(jmodel_ad.createData());
+    pinocchio::JointDataBase<JointDataAD> & jdata_ad_base = jdata_ad;
+
+    ConfigVector q(jmodel.nq());
+
+    ConfigVector lb(ConfigVector::Constant(jmodel.nq(), 0.));
+    ConfigVector ub(ConfigVector::Constant(jmodel.nq(), 1.));
```

Comment:
The only difference between generic `test` function and specialized one for `JointModelSplineTpl` is the configuration bound.
Is there way to specialize RandomConfigurationStep to handle that ?
If there is no way, can generate the bounds from `TestADOnJoints` visitor to avoid duplicating the code.